kura/org.eclipse.kura.rest.identity.provider/src/main/java/org/eclipse/kura/internal/rest/identity/provider/IdentityRestService.java

```diff
@@ -0,0 +1,207 @@
+        }
+    }
+
+    @POST
+    @RolesAllowed(REST_ROLE_NAME)
+    @Path("/identities")
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response createUser(final UserDTO userName) {
+        try {
+            logger.debug(DEBUG_MESSAGE, "createUser");
+            this.identityService.createUser(userName);
+        } catch (Exception e) {
+            throw DefaultExceptionHandler.toWebApplicationException(e);
+        }
+
+        return Response.ok().build();
+    }
+
+    @PUT
+    @RolesAllowed(REST_ROLE_NAME)
+    @Path("/identities")
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response updateUser(final UserDTO user) {
+        try {
+            logger.debug(DEBUG_MESSAGE, "updateUser");
+            this.identityService.updateUser(user);
+        } catch (Exception e) {
+            throw DefaultExceptionHandler.toWebApplicationException(e);
+        }
+
+        return Response.ok().build();
+    }
+
+    @POST
+    @RolesAllowed(REST_ROLE_NAME)
+    @Path("/identities/byName")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public UserDTO getUser(final UserDTO userName) {
+        try {
+            logger.debug(DEBUG_MESSAGE, "getUser");
+            return this.identityService.getUser(userName.getUserName());
+        } catch (Exception e) {
+            throw DefaultExceptionHandler.toWebApplicationException(e);
+        }
+
+    }
+
+    @DELETE
+    @RolesAllowed(REST_ROLE_NAME)
+    @Path("/identities")
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response deleteUser(final UserDTO userName) {
+        try {
+            logger.debug(DEBUG_MESSAGE, "deleteUser");
+            this.identityService.deleteUser(userName.getUserName());
+        } catch (Exception e) {
+            throw DefaultExceptionHandler.toWebApplicationException(e);
+        }
+
+        return Response.ok().build();
+    }
+
+    @GET
+    @RolesAllowed(REST_ROLE_NAME)
+    @Path("/defined-permissions")
+    @Produces(MediaType.APPLICATION_JSON)
+    public PermissionDTO getDefinedPermissions() {
+        try {
+            logger.debug(DEBUG_MESSAGE, "getDefinedPermissions");
+            return new PermissionDTO(this.identityService.getDefinedPermissions());
+        } catch (Exception e) {
+            throw DefaultExceptionHandler.toWebApplicationException(e);
+        }
+    }
+
+    @GET
+    @RolesAllowed(REST_ROLE_NAME)
+    @Path("/identities")
+    @Produces(MediaType.APPLICATION_JSON)
+    public UserConfigDTO getUserConfig() {
+        try {
+            logger.debug(DEBUG_MESSAGE, "getUserConfig");
+            UserConfigDTO userConfig = new UserConfigDTO();
+            userConfig.setUserConfig(this.identityService.getUserConfig());
+            return userConfig;
+        } catch (Exception e) {
+            throw DefaultExceptionHandler.toWebApplicationException(e);
+        }
+    }
+
+    @GET
+    @RolesAllowed(REST_ROLE_NAME)
```

Comment:
I think you can remove the RolesAllowed annotation an allow any user to access this.

Reply:
why ?